BST.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BST
+{
+    class BST
+    {
+
+        public Node root;
+
+
+        public BST()
+        {
+            root = null;
+        }
+
+        public void insert(int key)
+        {
+            root = insertRec(root, key);
+        }
+
+        public Node insertRec(Node root, int key)
+        {
+
+            if (root == null)
+            {
+                root = new Node(key);
+                return root;
+            }
+
+            if (key < root.key)
+                root.left = insertRec(root.left, key);
+            else if (key > root.key)
+                root.right = insertRec(root.right, key);
+
+            return root;
+        }
+
+        public void inorder()
+        {
+            inorderRec(root);
+        }
+
+        public void inorderRec(Node root)

```

Comment:
If you call a method only in your class, then it should be private. Why have something like 'SomethingRec()', when you can just have 'Something()'. Method name should describe what it does, not how it works.